src/go/configgenerator/listener_generator.go

```diff
@@ -54,113 +41,54 @@ const (
 
 // MakeListeners provides dynamic listeners for Envoy
 func MakeListeners(serviceInfo *sc.ServiceInfo) ([]*listenerpb.Listener, error) {
-	listener, err := makeListener(serviceInfo)
+	filterGenerators, err := MakeFilterGenerators(serviceInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	listener, err := MakeListener(serviceInfo, filterGenerators)
```

Comment:
How this will work in google3? I think our internal code will follow this flow:

1. Call `MakeFilterGenerator`
2. Override some of the key/value pairs in the slice. Remove SC v1, add our own filters.
3. Call `MakeListeners(serviceInfo, filterGenerators)`.

Is this correct?

Reply:
My thought is the filter_generator is responsible to generate the intended filter chains.